File: javascript/arreglos.js
let arreglo = Array.from({ length: 10 }, () => Math.floor(Math.random() * 100) + 1);
console.log("Arreglo inicial:", arreglo);

console.log("\n Recorrido con for clasico");
for (let i = 0; i < arreglo.length; i++) {
    console.log(`Posición ${i}: ${arreglo[i]}`);
}

console.log("\n Recorrido con for-each");
arreglo.forEach((valor) => {
    console.log(valor);
});

console.log("\n Cambiar valores impares por cero");
for (let i = 0; i < arreglo.length; i++) {
    if (arreglo[i] % 2 !== 0) {
        arreglo[i] = 0;
    }
}
console.log("Arreglo con impares en cero:", arreglo);

console.log("\n Multiplicar cada valor por su Indice");
for (let i = 0; i < arreglo.length; i++) {
    arreglo[i] = arreglo[i] * i;
}
console.log("Arreglo multiplicado por Indice:", arreglo);

function busquedaLineal(arr, objetivo) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === objetivo) {
            return i;
        }
    }
    return -1;
}

const valorBuscado = arreglo.length > 3 ? arreglo[3] : 0;
const posicion = busquedaLineal(arreglo, valorBuscado);
console.log(`\n Busqueda lineal del valor ${valorBuscado}: encontrado en posicion ${posicion}`);

File: go/arreglos.go
package main

import (
	"fmt"
	"math/rand"
)

func busquedaLineal(arr [10]int, objetivo int) int {
	for i := 0; i < len(arr); i++ {
		if arr[i] == objetivo {
			return i
		}
	}
	return -1
}

func main() {
	var arreglo [10]int
	for i := 0; i < len(arreglo); i++ {
		arreglo[i] = rand.Intn(100) + 1
	}
	fmt.Println("Arreglo inicial:", arreglo)

	fmt.Println("\n Recorrido con for clasico")
	for i := 0; i < len(arreglo); i++ {
		fmt.Printf("Posicion %d: %d\n", i, arreglo[i])
	}

	fmt.Println("\n Recorrido con for-each")
	for _, valor := range arreglo {
		fmt.Println(valor)
	}

	fmt.Println("\n Cambiar valores impares por cero")
	for i := 0; i < len(arreglo); i++ {
		if arreglo[i]%2 != 0 {
			arreglo[i] = 0
		}
	}
	fmt.Println("Arreglo con impares en cero:", arreglo)

	fmt.Println("\n Multiplicar cada valor por su Indice")
	for i := 0; i < len(arreglo); i++ {
		arreglo[i] = arreglo[i] * i
	}
	fmt.Println("Arreglo multiplicado por Indice:", arreglo)

	valorBuscado := arreglo[3]
	posicion := busquedaLineal(arreglo, valorBuscado)
	fmt.Printf("\nBusqueda lineal del valor %d: encontrado en posicion %d\n", valorBuscado, posicion)

}
